sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/FormRecognizerClientBuilder.java

```diff
@@ -232,6 +238,19 @@ public FormRecognizerClientBuilder credential(AzureKeyCredential apiKeyCredentia
         return this;
     }
 
+    /**
+     * Sets the {@link TokenCredential} used to authenticate HTTP requests.
+     *
+     * @param tokenCredential {@link TokenCredential} used to authenticate HTTP requests.
+     * @return The updated {@link FormRecognizerClientBuilder} object.
+     * @throws NullPointerException If {@code tokenCredential} is {@code null}.
+     */
+    public FormRecognizerClientBuilder credential(TokenCredential tokenCredential) {
+        Objects.requireNonNull(tokenCredential, "'tokenCredential' cannot be null.");
+        this.tokenCredential = tokenCredential;
```

Comment:
this.tokenCredential = Objects.requireNonNull(tokenCredential, "'tokenCredential' cannot be null.");